pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/defaultvalues/ATDefaultValues.java

```diff
@@ -4,76 +4,85 @@
-    public static final String ACTIVITY_CODE_REGISTRATION = "REGISTRATION";
-    public static final String COMPLETE = "COMPLETE";
-    public static final String GENOME_STUDY_FIELD_TYPE = "AT_GROUP_GENOME_STUDY";
-    private static final String GENOME_STUDY_CPT_ID = "GENOME_STUDY_CPT_ID";
-    private static final String PREFIX = "DDP_ATCP_";
-    private Dao dataAccess;
-
+    protected static final String EXIT_STATUS = "EXITSTATUS";
+    protected static final String AT_PARTICIPANT_EXIT = "AT_PARTICIPANT_EXIT";
+    protected static final String AT_GENOMIC_ID = "at_genomic_id";
+    protected static final String ACTIVITY_CODE_REGISTRATION = "REGISTRATION";
+    protected static final String COMPLETE = "COMPLETE";
+    protected static final String GENOME_STUDY_FIELD_TYPE = "AT_GROUP_GENOME_STUDY";
+    protected static final String GENOME_STUDY_CPT_ID = "GENOME_STUDY_CPT_ID";
+    protected static final String GENOMIC_ID_PREFIX = "DDP_ATCP_";
+
+    /**
+     * Inserts default data for a participant in the AT study: a genomic id and an exit status
+     *
+     * @return true if data was actually inserted
+     */
     @Override
     protected boolean setDefaultData() {
-        if (isParticipantDataNotInES()) {
-            throw new ESMissingParticipantDataException(
-                    String.format("Participant %s does not yet have profile and activities in ES",
-                    elasticSearchParticipantDto.getQueriedParticipantId()));
+        // expecting ptp has a profile and has completed the enrollment activity
+        if (elasticSearchParticipantDto.getProfile().isEmpty()
+                || elasticSearchParticipantDto.getActivities().isEmpty()) {
+            throw new ESMissingParticipantDataException(String.format("Participant %s does not yet have profile and "
+                    + "activities in ES", elasticSearchParticipantDto.getQueriedParticipantId()));
         }
 
-        boolean inserted = insertExitStatusForParticipant() && insertGenomicIdForParticipant();
-
+        Profile profile = elasticSearchParticipantDto.getProfile().orElseThrow();
+        String ddpParticipantId = profile.getGuid();
         try {
-            String ddpParticipantId = elasticSearchParticipantDto.getProfile().orElseThrow().getGuid();
-            ObjectTransformer objectTransformer = new ObjectTransformer(instance.getName());
-            this.setDataAccess(new ParticipantDataDao());
             List<ParticipantData> participantDataList =
-                    ((ParticipantDataDao) dataAccess).getParticipantDataByParticipantId(ddpParticipantId);
+                    participantDataDao.getParticipantDataByParticipantId(ddpParticipantId);
+
+            boolean hasGenomeId = participantDataList.stream().anyMatch(
+                    participantDataDto -> ATDefaultValues.GENOME_STUDY_FIELD_TYPE.equals(
+                            participantDataDto.getRequiredFieldTypeId()));
+            if (!hasGenomeId) {
```

Comment:
I think I see it...we were creating more and more genomic ids each time we had a path through `generateDefaults()`.  Yowza!